build_scripts/build_collect_diff.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+#
+# Script to build the kernel, collect compiled file lists using modified kernel scripts,
+# and generate a git diff report based on the collected lists.
+
+set -e
+
+# check and install gcc-11 if not already installed
+install_package_safe() {
+    if ! command -v libssl-dev &> /dev/null; then
+        sudo apt-get update
+        sudo apt-get install -y libssl-dev
+    else
+        echo "libssl-dev is already installed."
+    fi
+}
+
+# safely apply a patch to linux kernel
+apply_patch() {
+    # shellcheck disable=SC2154
+    local patch_path="$root_path/scripts/change-impact-tools/fixdep-patch.file"
+
+    # Stash any changes if there is any
+    if ! git diff --quiet; then
+        git stash
+    fi
```

Comment:
If you stash a user's changes without telling them, they are going to be surprised and perhaps a little unhappy that their repository is not in the state they expect. Instead of stashing the changes, you might consider simply exiting your script with a descriptive error if the repository is not in a state that you feel is amenable to analyzing. The user can then decide how to manage their changes and re-run your script at a later time.

Reply:
Yes. I have fixed this problem.